tensorboard/plugins/hparams/tf_hparams_utils/tf-hparams-utils.html

```diff
@@ -124,42 +155,131 @@
-      // 'columnIndex'.
+      // Returns sessionGroup's column value of the column with index
+      // 'columnIndex' in schema.
       function columnValueByIndex(schema, sessionGroup, columnIndex) {
-        if (columnIndex < schema.hparamInfos.length) {
+        if (columnIndex < schema.hparamColumns.length) {
           return hparamValueByIndex(schema, sessionGroup, columnIndex);
         }
-        return metricValueByIndex(schema, sessionGroup,
-                                  columnIndex - schema.hparamInfos.length);
+        return metricValueByIndex(
+            schema, sessionGroup,
+            columnIndex - schema.hparamColumns.length);
       }
       utils.columnValueByIndex = columnValueByIndex;
 
       // Returns an array [min, max] representing the minimum and maximum
       // value of the given column in the sessionGroups array.
       // Ignores session groups with missing values for the column.
       function numericColumnExtent(schema, sessionGroups, columnIndex) {
-        return d3.extent(sessionGroups,
-                         sg=>columnValueByIndex(schema, sg, columnIndex));
+        return d3.extent(
+            sessionGroups,
+            sg=>columnValueByIndex(schema, sg, columnIndex));
       }
       utils.numericColumnExtent = numericColumnExtent;
 
+      // Returns true if the column indexed by 'columnIndex' is currently
+      // displayed (has 'displayed' in its {hparam, metric}Columns entry).
+      function isColumnDisplayed(schema, columnIndex) {
+        if (columnIndex < schema.hparamColumns.length) {
+          return schema.hparamColumns[columnIndex].displayed;
+        }
+        return schema.metricColumns[columnIndex - schema.hparamColumns.length]
+            .displayed;
+      }
+      utils.isColumnDisplayed = isColumnDisplayed;
+
+      // Converts a visibleSchema columnIndex to a schema columnIndex.
+      // Returns the schema-relative columnIndex for the visible column with
+      // schema.visible-releative columnIndex given by 'visibleColumnIndex'.
+      function getAbsoluteColumnIndex(schema, visibleColumnIndex) {
+        let result;
+        if (visibleColumnIndex < schema.visibleSchema.hparamInfos.length) {
+          result = schema.hparamColumns.findIndex(
+              c => (c.hparamInfo.name ===
+                    schema.visibleSchema.hparamInfos[visibleColumnIndex].name));
+        } else {
+          const metricIndex =
+                visibleColumnIndex - schema.visibleSchema.hparamInfos.length;
+          const metricName = schema.visibleSchema.metricInfos[metricIndex].name;
+          result = schema.hparamColumns.length + schema.metricColumns.findIndex(
+              c => (c.metricInfo.name === metricName));
+        }
+        console.assert(result !== undefined);
+        console.assert(result !== -1);
+        console.assert(isColumnDisplayed(schema, result));
+        return result;
+      }
+      utils.getAbsoluteColumnIndex = getAbsoluteColumnIndex;
```

Comment:
If I were you, I'd convert this util file to TypeScript and do without nested functions and assignments like this.